client/src/graphql/queries.js

```diff
@@ -130,5 +130,13 @@ export default {
         _id
       }
     }
+  `,
+  FETCH_SHOP_COFFEES: gql`
```

Comment:
I don't really get why this has to exist, when we fetch shops don't we already return all the coffees that shop has?

Reply:
If I could just have a little exposition about this one issue then I'll be happy to approve this PR.

Reply:
So I wrote a backend query so this query can use, basically when a filter was provided, this should only fetch the coffess which belongs to that shop and meet the filter conditions. So basically, I need to refetch each time based on the filter, so the page will dynamically change based on filter selections. And I don't want to modify the original one (fetch coffees from store) in case you need to test something